src/circuit_proof/verifier.rs

```diff
@@ -303,21 +282,21 @@ impl<'a> VerifierCS<'a> {
                 .chain(x_inv_sq.iter().cloned()) // ipp_proof.R_vec
                 .chain(V_coeff) // V
                 .chain(T_scalars.iter().cloned()), // T_points
-            iter::once(&A_I)
-                .chain(iter::once(&A_O))
-                .chain(iter::once(&W_L_point))
-                .chain(iter::once(&W_R_point))
-                .chain(iter::once(&W_O_point))
-                .chain(iter::once(&S))
-                .chain(iter::once(&gens.pedersen_gens.B))
-                .chain(iter::once(&gens.pedersen_gens.B_blinding))
-                .chain(gens.G(n))
-                .chain(gens.H(n))
-                .chain(Ls.iter())
-                .chain(Rs.iter())
-                .chain(Vs.iter())
-                .chain(T_points.iter()),
-        );
+            iter::once(Some(A_I))
+                .chain(iter::once(Some(A_O)))
+                .chain(iter::once(Some(W_L_point)))
+                .chain(iter::once(Some(W_R_point)))
+                .chain(iter::once(Some(W_O_point)))
+                .chain(iter::once(Some(S)))
+                .chain(iter::once(Some(gens.pedersen_gens.B)))
+                .chain(iter::once(Some(gens.pedersen_gens.B_blinding)))
+                .chain(gens.G(n).map(|&g_i| Some(g_i)))
+                .chain(gens.H(n).map(|&h_i| Some(h_i)))
```

Comment:
These might be `G_i`, `H_i` if we're using capitals for the points and `g_i`, `h_i` for their scalars?

Reply:
👍